src/ui/win32/bindings/MemoryViewerControlBinding.cpp

```diff
@@ -236,6 +239,68 @@ bool MemoryViewerControlBinding::HandleNavigation(UINT nChar)
     }
 }
 
+bool MemoryViewerControlBinding::HandleShortcut(UINT nChar)
+{
+    const bool bShiftHeld = (GetKeyState(VK_SHIFT) < 0);
+    const bool bControlHeld = (GetKeyState(VK_CONTROL) < 0);
+
+    switch (nChar)
+    {
+        case 'C':
+            if (bControlHeld)
+            {
+                const auto& pEmulatorContext = ra::services::ServiceLocator::Get<ra::data::context::EmulatorContext>();
+                const auto nValue = pEmulatorContext.ReadMemory(m_pViewModel.GetAddress(), m_pViewModel.GetSize());
+                std::wstring sValue = ra::data::MemSizeFormat(nValue, m_pViewModel.GetSize(), MemFormat::Hex);
+
+                ra::services::ServiceLocator::Get<ra::services::IClipboard>().SetText(ra::Widen(sValue));
+            }
+            return true;
+
+        case 'V':
+            if (bControlHeld and !m_pViewModel.IsReadOnly())
+            {
+                auto nAddress = m_pViewModel.GetAddress();
+                const auto& pEmulatorContext = ra::services::ServiceLocator::Get<ra::data::context::EmulatorContext>();
+                std::wstring sClipboardText = ra::services::ServiceLocator::Get<ra::services::IClipboard>().GetText();
+                const auto nNibblesForSize = ra::data::MemSizeBytes(m_pViewModel.GetSize());
+                std::vector<std::uint32_t> v_nValues;
+                std::wstring sError;
+
+                if (sClipboardText.empty())
+                    return false;
+
+                // Split the clipboard value into substrings matching the current size and check if they're valid hexadecimal values
```

Comment:
Recommend documenting the shift+paste functionality, as it's something that is not normal and won't be obvious when looking at this code some time in the future.
```
// If the clipboard text is smaller than a single write, it will be treated as a padded value:
//    C => 0C, 000C, or 0000000C (depending on currently selected viewer size)
// If shift is held, and the clipboard string is bigger than a single write, multiple writes will occur. 
// If the last write is not a complete chunk, it will be padded in the same way as described above:
//    12345678C => 12 34 56 78 0C, 1234 5678 000C, or 12345678 0000000C
```